fusestore/store/src/executor/meta_handlers.rs

```diff
@@ -91,51 +110,101 @@ impl RequestHandler<DropDatabaseAction> for ActionHandler {
         &self,
         act: DropDatabaseAction,
     ) -> common_exception::Result<DropDatabaseActionResult> {
-        let mut meta = self.meta.lock();
-        let _ = meta.drop_database(&act.plan.db, act.plan.if_exists)?;
-        Ok(DropDatabaseActionResult {})
+        let cr = LogEntry {
+            txid: None,
+            cmd: DropDatabase { name: act.plan.db },
+        };
+
+        let rst = self
+            .meta_node
+            .write(cr)
+            .await
+            .map_err(|e| ErrorCode::MetaNodeInternalError(e.to_string()))?;
+
+        match rst {
+            AppliedState::DataBase { .. } => Ok(DropDatabaseActionResult {}),
+            _ => Err(ErrorCode::MetaNodeInternalError("not a Database result")),
+        }
     }
 }
-// table
 
+// table
 #[async_trait::async_trait]
 impl RequestHandler<CreateTableAction> for ActionHandler {
     async fn handle(
         &self,
         act: CreateTableAction,
     ) -> common_exception::Result<CreateTableActionResult> {
         let plan = act.plan;
-        let db_name = plan.db;
-        let table_name = plan.table;
+        let db_name = &plan.db;
+        let table_name = &plan.table;
+        let if_not_exists = plan.if_not_exists;
 
-        info!("create table: {:}: {:?}", db_name, table_name);
+        info!("create table: {:}: {:?}", &db_name, &table_name);
 
-        let mut meta = self.meta.lock();
+        // if database not exists, return early.
+        let db = self.meta_node.get_database(db_name).await.ok_or_else(|| {
+            ErrorCode::UnknownDatabase(format!("create table: database not found {:}", db_name))
+        })?;
```

Comment:
For the same reason, a just-created-db may not be seen on this node. Let the state machine do this check.